### PR TITLE
Updated character controller with CRUD operations

### LINKED ISSUE #39: [DEV] Character Controller

### PR DESCRIPTION
**Pull request by briiitelord**

### CODE REVIEW

File: src/main/java/com/amplifier/controllers/CharacterController.java [hunk 1]
```diff
@@ -1,5 +1,76 @@
...
+    //RUD by ID
+
+    @ApiOperation(value = "Find character by id number", notes = "Provide an id to lookup a specific character from the API", response = Character.class)
+    @GetMapping(path = "/character/{id}")
+    public @ResponseBody Character getCharacterById(@RequestParam(value = "id", name = "id") int id) {
+        // System.out.println("TEST: " + characterService.getCharacterById(id));
+        return characterService.getCharacterById(id);
+    }
+
+    @PutMapping("/character/{Id}")
```
Comment (FullSol) [inline src/main/java/com/amplifier/controllers/CharacterController.java]: You did this correct per the current documentation. I am going to be changing the documentation to a query. That means the URI will be "/character?id={Id}"

File: src/main/java/com/amplifier/controllers/CharacterController.java [hunk 2]
```diff
@@ -1,5 +1,76 @@
...
+    @ApiOperation(value = "Update character entity by character Id")
+    public @ResponseBody Character updateCharacter(@RequestBody int Id) {
+        return characterService.updateCharacterById(Id) ? "UPDATE_SUCCESSFUL" : "UPDATE_FAILED";
+    }
+
+   
+
+    @DeleteMapping("/character/{Id}")
```
Comment (FullSol) [inline src/main/java/com/amplifier/controllers/CharacterController.java]: You did this correct per the current documentation. I am going to be changing the documentation to a query. That means the URI will be "/character?id={Id}"

File: src/main/java/com/amplifier/controllers/CharacterController.java [hunk 3]
```diff
@@ -1,5 +1,76 @@
...
+    @PostMapping("/character")
+    @ApiOperation(value = "Create new character entity")
+    public @ResponseBody Character createCharacter(@RequestBody Character character) {
+        return characterService.createCharacter(character) ? "CREATION_SUCCESSFUL" : "CREATION_FAILED";
+    }
+
+    //RUD by ID
+
+    @ApiOperation(value = "Find character by id number", notes = "Provide an id to lookup a specific character from the API", response = Character.class)
+    @GetMapping(path = "/character/{id}")
```
Comment (FullSol) [inline src/main/java/com/amplifier/controllers/CharacterController.java]: You did this correct per the current documentation. I am going to be changing the documentation to a query. That means the URI will be "/character?id={Id}"

**Review by FullSol — CHANGES_REQUESTED** [review]
So, you did this absolutely correct, and I apologize for the request change. This is due to the way we first decided to write the documentation.

File: src/main/java/com/amplifier/controllers/CharacterController.java [hunk 4]
```diff
@@ -1,5 +1,76 @@
...
+@Api(value = "CharacterRestController", description = "REST controller related to Character Entities")
 public class CharacterController {
 
+    @Autowired
+    private CharacterService characterService;
+
+    //Get all characters
+
+    @GetMapping("/characters")
+    @ApiOperation(value = "Find all characters")
```
Comment (pyaeger) [inline src/main/java/com/amplifier/controllers/CharacterController.java]: Please add notes to ApiOperation documentation and return object in ApiOperation as well if a return type is present.

File: src/main/java/com/amplifier/controllers/CharacterController.java [hunk 5]
```diff
@@ -1,5 +1,76 @@
...
+    public @ResponseBody List<Character> getAllCharacters() {
+        return characterService.getAllCharacters();
+    }
+
+    //Create character
+
+    @PostMapping("/character")
+    @ApiOperation(value = "Create new character entity")
```
Comment (pyaeger) [inline src/main/java/com/amplifier/controllers/CharacterController.java]: Please add ApiOperation note and return where relevant. Thanks!

**Review by pyaeger — CHANGES_REQUESTED** [review]
Just a few little changes requested. Thanks!

**Review by FullSol — APPROVED** [review]

**Review by pyaeger — APPROVED** [review]
Looks good.

**Comment by pyaeger** [comment]
Conflicting files need to be resolved before I can merge this pull request.

**Comment by FullSol** [comment]
@pyaeger Yeah, this is a process that is really annoying. Go ahead and download GitHub desktop and we will go over this when we get a chance today.

**Pull request merged**